### ISSUE TITLE
[SimplerEnv] Question about reproducing results with the released StarVLA-OFT checkpoints

### ISSUE THREAD

**Issue by jzndd** [starVLA/starVLA#424]
Hi StarVLA team,

Thank you for open-sourcing StarVLA and the pretrained checkpoints! I have been evaluating the released Bridge + RT-1 checkpoints on SimplerEnv. The π and GR00T variants reproduce reasonably closely, but I am seeing a much larger gap for both OFT checkpoints.

I may have missed an OFT-specific evaluation setting, so I would really appreciate any guidance.

## Setup

- GPU: NVIDIA A100
- StarVLA stable branch:
  `3422b9f2387b6f682cf02802904a77b23ab13afd`
- SimplerEnv:
  `06accaca93535902d408da4855f21cece12bceb7`
- Python 3.10
- PyTorch 2.6.0 + CUDA 12.4
- Transformers 4.57.0
- NumPy 1.24.4 in the SimplerEnv environment
- BF16 policy inference
- Official StarVLA SimplerEnv model client and visual-matching task configurations
- Default adaptive action ensembling:
  - WidowX horizon: 7
  - Google Robot horizon: 2
  - `adaptive_ensemble_alpha=0.1`
- Correct normalization keys were selected per embodiment:
  - WidowX: `oxe_bridge`
  - Google Robot: `oxe_rt1`

For WidowX, I evaluated 96 episodes per task, for 384 episodes in total. For Google Robot visual matching, I evaluated the complete 864-episode task matrix.

All evaluation jobs completed successfully without inference or simulator errors.

The only checkpoint configuration changes were replacing the original local `base_vlm` paths with the downloaded Hugging Face model paths. For the legacy Qwen2.5-OFT checkpoint, I also mapped the dataset-statistics keys as follows:

- `bridge_dataset` → `oxe_bridge`
- `fractal20220817_data` → `oxe_rt1`

## WidowX results

| Released checkpoint | Public checkpoint step | Reported / model-zoo result | My result |
|---|---:|---:|---:|
| Qwen2.5-PI-Bridge-RT-1 | 30K | 62.5% | 63.54% |
| Qwen2.5-GR00T-Bridge-RT-1 | 30K | 63.6% | 59.11% |
| Qwen2.5-OFT-Bridge-RT-1 | 10K | 41.8% | **9.38%** |
| Qwen3VL-PI_v3-Bridge-RT-1 | 50K | 69.8% | 69.53% |
| Qwen3VL-GR00T-Bridge-RT-1 | 20K | 65.3% | 60.16% |
| Qwen3VL-OFT-Bridge-RT-1 | 5K | 42.7% in the model zoo | **26.04%** |

The per-task OFT results are:

| Checkpoint | Spoon | Carrot | Stack cube | Eggplant | Average |
|---|---:|---:|---:|---:|---:|
| Qwen2.5-OFT, step 10K | 4.17% | 16.67% | 0.00% | 16.67% | 9.38% |
| Qwen3VL-OFT, step 5K | 20.83% | 12.50% | 8.33% | 62.50% | 26.04% |

## Google Robot visual-matching result

For the released Qwen3VL-OFT step-5K checkpoint, I obtained:

| Task | My result | Reported result |
|---|---:|---:|
| Pick Coke Can | 62.00% | 95.3% |
| Move Near | 51.25% | 75.0% |
| Open/Close Drawer | 26.39% | 68.8% |
| Open Top Drawer and Place Apple | 0.00% | 66.1% |
| Macro average | **34.91%** | **76.0%** |

I understand that the reported Google Robot score may have been produced by a different OFT checkpoint, so this comparison may not be checkpoint-aligned.

## Possible checkpoint mismatch

I noticed a possible discrepancy between the benchmark page and the released checkpoint:

- The benchmark page reports **StarVLA-OFT (Qwen3-VL), step 65K, 64.6%** on WidowX.
- The released `StarVLA/Qwen3VL-OFT-Bridge-RT-1` repository currently contains a **step-5K checkpoint**.
- The model zoo lists **42.7%** for this released checkpoint.

References:

- Benchmark page: https://starvla.github.io/docs/benchmarks/simplerenv/
- Released checkpoint: https://huggingface.co/StarVLA/Qwen3VL-OFT-Bridge-RT-1

I would be happy to provide the full per-task CSV, configs, logs, or example rollout videos if they would help with debugging.

Thanks again for the great work and for making the checkpoints available!

**Comment by JJLibra** [comment]
I think it mainly comes from a checkpoint mismatch. The paper reports the Qwen3-VL OFT result using a much longer-trained checkpoint (e.g., 65K steps on WidowX), while the public checkpoint is only 5K steps. Therefore, the released checkpoint is unlikely to reproduce the paper-level performance. It would be helpful if the authors could clarify and release the exact checkpoints used for the reported results.

**Comment by JinhuiYE** [comment]
Hi @jjlibr and everyone,

Thank you for the thorough reproduction and diagnosis, and thanks @JJLibra for the follow-up. Your analysis is very helpful.

We agree with the assessment: the performance gap is likely due to a checkpoint mismatch between the publicly released OFT models and those used for the reported results.

We will:

Conduct an internal check to verify the checkpoint versions and step discrepancies (e.g., 5K vs. 65K) across our release records and benchmark logs.

Add a detailed Model Card to each released checkpoint on Hugging Face, specifying training steps, data mixture, expected benchmark performance, and known limitations to avoid future confusion.

We will update this issue once the review is complete. Please let us know if you have any other questions.

Thanks again for your valuable feedback!

**Comment by pxlae** [comment]
@JinhuiYE , @jzndd : 

I fully agree with this and to add on top of this I think there is a huggingface 20k checkpoint. I validated on below is a screenshot. 

<img width="513" height="211" alt="Image" src="https://github.com/user-attachments/assets/70850b51-2d61-4c99-a83d-2e51bf24f9c2" />

Few things to note on the benchmarking that I have noticed. 
1. Reproduction is not supported well for at least simpler env. There could be few improvements like seeding and so on. 
2. Different seeds give different numbers
3. Reporting for SimplerEnv should be consistent with litrature to some extent (i.e : VM,VW seperation , GR/WidowX seperation) 

I can colloaborate on some of the thigns above if support is needed.

**Comment by krystal21** [comment]
Hi @JinhuiYE, just following up—are there any updates on the checkpoint review and Model Cards mentioned above? I encountered the same issue as well. Thanks!

**Comment by JinhuiYE** [comment]
> [@JinhuiYE](https://github.com/JinhuiYE) , [@jzndd](https://github.com/jzndd) :
> 
> I fully agree with this and to add on top of this I think there is a huggingface 20k checkpoint. I validated on below is a screenshot.
> 
> <img alt="Image" width="513" height="211" src="https://private-user-images.githubusercontent.com/141075920/624855312-70850b51-2d61-4c99-a83d-2e51bf24f9c2.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODYwNjk2NjksIm5iZiI6MTc4NjA2OTM2OSwicGF0aCI6Ii8xNDEwNzU5MjAvNjI0ODU1MzEyLTcwODUwYjUxLTJkNjEtNGM5OS1hODNkLTJlNTFiZjI0ZjljMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwODA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDgwN1QwMjIyNDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hZTBhNGU3M2M4MjhiODAyMGU3M2U1OTRhNGVhMzIzMzE0YmNmZTUzZmM4NDQ5YWZmOWRiNTQwMDI2MWYwYzBiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.zRk2pcVclrYaEFHt-829DmsDohwCNtP93pJWR49uhm0">
> Few things to note on the benchmarking that I have noticed.
> 
> 1. Reproduction is not supported well for at least simpler env. There could be few improvements like seeding and so on.
> 2. Different seeds give different numbers
> 3. Reporting for SimplerEnv should be consistent with litrature to some extent (i.e : VM,VW seperation , GR/WidowX seperation)
> 
> I can colloaborate on some of the thigns above if support is needed.

Thanks for your valuable feedback – and apologies for missing your previous messages. I've been juggling quite a few things recently, so I'm afraid they slipped through.

We fully agree on the issues you raised (reproducibility, seed variance, and reporting alignment for SimpleEnv). We recognize that support is still lacking, especially in simpler environments.

We strongly welcome and need PRs on these aspects to improve our codebase and benchmarking. Happy to collaborate if you need any support from our side.

Thanks again, and we look forward to your contributions!

**Comment by JinhuiYE** [comment]
> Hi [@JinhuiYE](https://github.com/JinhuiYE), just following up—are there any updates on the checkpoint review and Model Cards mentioned above? I encountered the same issue as well. Thanks!


We found that the checkpoint we uploaded earlier was actually incorrect, and we removed the original files. We are currently re‑training the model from scratch and will update the checkpoint and Model Cards as soon as we have the new results ready.

I'll make sure to ping you here once everything is updated. Thanks again for bearing with us!

**Comment by CoffeeDrivenCoder** [comment]
Hi @JinhuiYE, I would like to contribute a focused reproducibility improvement
for this issue.

My proposed scope is:

- add an explicit evaluation seed option to the SimplerEnv entrypoint;
- seed the relevant Python, NumPy, PyTorch, and environment RNGs where supported;
- record the seed, checkpoint, environment, task, episode count, and success rate
  in the evaluation summary;
- add deterministic unit coverage for the new helper without requiring a full
  GPU benchmark run.

I will keep this change limited to the StarVLA evaluation scripts and will not
modify checkpoints or change the reported benchmark results.

Does this scope match what the maintainers would like to see?

**Comment by JinhuiYE** [comment]
Thanks for proposing a focused scope. Yes, the seed and evaluation-provenance changes described here are appropriate for a PR: keep them isolated to the StarVLA SimplerEnv evaluation path, expose the seed explicitly, seed each supported RNG, and record the checkpoint, environment/task, episode count, seed, and success rate in the output. A CPU-testable deterministic helper is also useful. To keep the boundary clear, this improves reproducibility and reporting but does not resolve or validate the released OFT checkpoint mismatch; that artifact question remains open separately in this issue.
